### ISSUE TITLE
Facing 'ffi_c' load error with latest chef 16

### ISSUE THREAD

**Issue by NAshwini** [chef/chef#10878]
## Description
While running Chef-client with any of the simple cookbook getting below error:
```
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in 
require': cannot load such file -- ffi_c
```
## Chef Version

16.9.20

## Platform Version

Windows

## Replication Case

Got error while running chef-client with an simple cookbook and this was also reproduce while installing azure-chef-extension.

## Client Output

```

```

## Stacktrace

```
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in 
require': cannot load such file -- ffi_c
```

**Comment by jcleavel** [comment]
I am also having the same issue.

I tried bootstrapping a new node (via the Azure Chef VM Extension) with Chef Client versions `v16.9.20` and `v16.8.14` and received the same failure both times.

My nodes are running Windows Server Core 2019 `v10.0.17763` (Build 1809).

Running the following command fixes the problem for that node, but it is not something that I can insert into my automated node provisioning process.

```powershell
 C:\opscode\chef\embedded\bin\gem.exe install ffi --platform=ruby
```

Here is a failed bootstrap log (via the Azure Chef VM extension) with full stack traces:

```
[01/13/2021 19:33:54.76] Executing: C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\enable.cmd 
[01/13/2021 19:37:52.27] Execution Complete.
######
Execution Output:
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>set CHEF_EXT_DIR=C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\ 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>echo C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\ 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>REM Installing chef-client 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>powershell -nologo -noprofile -executionpolicy unrestricted Import-Module C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\bin\chef-install.psm1;Install-ChefClient 
Environment variables not passed.
Checking Chef Client ...
Set CHEF_LICENSE Environment variable as accept
Downloading Chef Client ...
ModuleType Version    Name                                ExportedCommands                                             
---------- -------    ----                                ----------------                                             
Script     0.0        Omnitruck                           {Get-ProjectMetadata, Install-Project, install}              
Installing chef from C:\windows\TEMP\chef-client-16.9.20-1-x64.msi
[01/13/2021 19:37:47] Installing Azure-Chef-Extension gem
Successfully installed azure-chef-extension-0.0.1
1 gem installed
[01/13/2021 19:37:51] Installed Azure-Chef-Extension gem successfully
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>REM set envioronment variable CHEF_LICENSE to accept-no-persist if not defined (reference: https://docs.chef.io/chef_license_accept). 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>IF NOT DEFINED CHEF_LICENSE set CHEF_LICENSE=accept-no-persist 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>set path=C:\opscode\chef\bin;C:\opscode\chef\embedded\bin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\Docker;C:\Program Files\PowerShell\7\;C:\windows\system32\config\systemprofile\AppData\Local\Microsoft\WindowsApps 
C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1>ruby C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\bin\chef-enable.rb 


Execution Error:
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require': cannot load such file -- ffi_c (LoadError)
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/ffi-1.14.2-x64-mingw32/lib/ffi.rb:6:in `rescue in <top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/ffi-1.14.2-x64-mingw32/lib/ffi.rb:3:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api/security.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api/security.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/security.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/security.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control/windows.rb:20:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control/windows.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:30:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:30:in `<class:FileAccessControl>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:27:in `<class:Chef>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:22:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/mixin/enforce_ownership_and_permissions.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/mixin/enforce_ownership_and_permissions.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider.rb:22:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider.rb:22:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/execute.rb:20:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/execute.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/script.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/script.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/windows_script.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/windows_script.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/batch.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/batch.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/providers.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/providers.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef.rb:25:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef.rb:25:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `rescue in require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:147:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/azure-chef-extension-0.0.1/lib/chef/azure/commands/enable.rb:3:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `rescue in require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:147:in `require'
	from C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/1210.13.4.1/bin/chef-enable.rb:4:in `<main>'
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:94:in `require': 126: The specified module could not be found.   - C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/ffi-1.14.2-x64-mingw32/lib/2.7/ffi_c.so (LoadError)
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:94:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/ffi-1.14.2-x64-mingw32/lib/ffi.rb:4:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:146:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api/security.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/api/security.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/security.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/win32/security.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control/windows.rb:20:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control/windows.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:30:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:30:in `<class:FileAccessControl>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:27:in `<class:Chef>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/file_access_control.rb:22:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/mixin/enforce_ownership_and_permissions.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/mixin/enforce_ownership_and_permissions.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider.rb:22:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider.rb:22:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/execute.rb:20:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/execute.rb:20:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/script.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/script.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/windows_script.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/windows_script.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/batch.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/provider/batch.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/providers.rb:19:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef/providers.rb:19:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef.rb:25:in `require_relative'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.9.20-universal-mingw32/lib/chef.rb:25:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `rescue in require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:147:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/azure-chef-extension-0.0.1/lib/chef/azure/commands/enable.rb:3:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `rescue in require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:147:in `require'
	from C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/1210.13.4.1/bin/chef-enable.rb:4:in `<main>'
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require': cannot load such file -- chef (LoadError)
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require'
	from C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/azure-chef-extension-0.0.1/lib/chef/azure/commands/enable.rb:3:in `<top (required)>'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:159:in `rescue in require'
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:147:in `require'
	from C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/1210.13.4.1/bin/chef-enable.rb:4:in `<main>'
C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require': cannot load such file -- chef/azure/commands/enable (LoadError)
	from C:/opscode/chef/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require'
	from C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/1210.13.4.1/bin/chef-enable.rb:4:in `<main>'

######
 Number of Tries:
 1Command C:\Packages\Plugins\Chef.Bootstrap.WindowsAzure.ChefClient\1210.13.4.1\enable.cmd of Chef.Bootstrap.WindowsAzure.ChefClient has exited with Exit code: 1
```

**Comment by darwinjames** [comment]
Workaround: "bootstrap_version": "16.8.14". It works.

**Comment by jcleavel** [comment]
That was the first thing I tried @darwinjames. It didn't work for my particular infra configuration, unfortunately.

**Comment by tas50** [comment]
We've identified a problem that is resulting in the newer version of FFI coming in, which we hope to have fixed tonight.

**Comment by tas50** [comment]
This should be resolved in the 16.9.29 release. Feel free to file another issue if you're still seeing issues in that build.
